megatron/data/biencoder_dataset_utils.py

```diff
@@ -146,11 +148,13 @@ def get_block_samples_mapping(block_dataset, title_dataset, data_prefix, num_epo
         indexmap_filename += '_1sentok'
     indexmap_filename += '.npy'
 
+    indexmap_file_path = str(Path(get_args().save).joinpath(indexmap_filename).absolute())
```

Comment:
`resolve` is usually recommended over `absolute`. Also it should be fine to leave as a Path, and we could keep the variable `indexmap_filename` to reduce the diff

Reply:
(Same with other files)